inc/sairouterinterface.h

```diff
@@ -316,6 +316,19 @@ typedef enum _sai_router_interface_attr_t
      */
     SAI_ROUTER_INTERFACE_ATTR_SELECTIVE_COUNTER_LIST,
 
+    /**
+     * @brief Attribute used to specify external My MAC entry that will
+     * be used in place of any implicit entry created during RIF processing
+     * for this {port, vlan, MAC address}
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_MY_MAC
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_ROUTER_INTERFACE_ATTR_MY_MAC,
+
```

Comment:
@erohsik  

1.  If SAI\_ROUTER\_INTERFACE\_ATTR\_MY\_MAC is not null, does it mean that the device must not create a termination entry with MAC == SAI\_ROUTER\_INTERFACE\_ATTR\_SRC\_MAC\_ADDRESS?
2.  What is the meaning of binding a RIF to MY\_MAC entry? Does it mean that the MY\_MAC entry takes effect only for packets that ingress on this RIF? I am trying to see how it relates to SAI\_MY\_MAC\_ATTR\_PORT/VLAN\_ID
3.  And lastly, why not achieve this by having a SAI\_ROUTER\_INTERFACE\_ATTR\_NO\_TERM\_MAC of type boolean to achieve the goal listed in the heading.

Reply:
Thanks. I prefer your original model over the suggestion. I think all NOS would anyway maintain this reference counting;  we can avoid adding unnecessary state in the SAI implementation.

 Further, the suggested model gives a impression that the pointed to MyMAC entry should only be used by the RIFs that point to this MyMAC entry and vice versa. Today, one RIF can be terminated by more than one MyMAC entries. One MyMAC entry (if we skip SAI\_MY\_MAC\_ATTR\_PORT/VLAN\_ID) can terminate multiple RIFs.

@JaiOCP @itaibaz Can you please give your feedback on the above. 

Today, SAI\_ROUTER\_INTERFACE\_ATTR\_SRC\_MAC\_ADDRESS is create-and-set, however SAI\_ROUTER\_INTERFACE\_ATTR\_MY\_MAC is create-only. Any reasons?

Reply:
@rck-innovium Even with this proposal many RIF can point to the same my mac object, right?